python/heterocl/devices.py

```diff
@@ -59,11 +75,15 @@ def __repr__(self):
 
 class Memory(object):
     """The base class for memory modules"""
-    def __init__(self, types, cap, channels=None):
+    def __init__(self, types, capacity=0, channels=0, port=0):
+        # memory device type (e.g., DRAM, HBM)
         self.types = types
-        self.capacity = cap
+        # memory maximum capacity per-bank
+        self.capacity = capacity
```

Comment:
What do you think if we add an optional argument "uint"? By default we use GB. For smaller memory we use `Memory(type=..., capacity=32, unit="KB")`

Reply:
No. We should try to limit the number of arguments. Maybe we should just remove `capacity` since it is not really used now.

Reply:
I think we'd better use it for checking. Otherwise the generated code would not run if the buffer size is greater than the capacity. E.g., XRT will complain that the memory is our of limit and cannot be allocated.

Reply:
I will add the checking logic in .to(). It would be just a few lines of code

Reply:
Then we should just stick to one unit.